app/controllers/bus_stops_controller.rb

```diff
@@ -83,13 +83,35 @@ def update
     @stop.decide_if_completed_by current_user
     if @stop.save
       flash[:notice] = 'Bus stop was updated.'
-      redirect_to bus_stops_path
+      if params[:commit] == 'Save and next' && params[:route_id]
+        route_id = params[:route_id]
+        next_stop = Route.find(route_id)
+                         .next_stop_in_sequence(@stop, route_id, params[:direction])
+        if next_stop
+          redirect_to(action: 'edit',
```

Comment:
Shouldn't need the parentheses around either `redirect_to`, which I mention because we don't usually write them.